### ISSUE TITLE
Feature Request: individual quantities/units for different packaging sizes without using specific predefined QU

### ISSUE THREAD

**Issue by DrAColian** [grocy/grocy#2414]
Hey berrnd,

I've done some searching and found topics closely related, but not, what I have in mind. So if it's a duplicate, I'm sorry. (If not, I would be happy, if you would consider this idea! :) )

So, I have several products, that come in different sizes/units. I will use a sepecific generic example of water. The example is not covering everything (so it's specific), but it can be adapted to any other product that has a similar behaviour (flour, mayonnaise, Wiener Würstchen :D)

I can buy water in packs, bottles and cans. It can be packed as 10 packs, 4 bottles, 12 cans. A bottle can have 0.75l, 1.5l, 2l, whereas a can can be 0.25l, 0.33l or 0.5l.

So, especially in comparing prices per liter, it's nice to get the individual amounts in l per packaging unit. Right now, I add all those units manually, but I think it's not the best way to use it.

That leads to my feature request: It would be nice to just have "general-SI-QU" (like can, bottle, glass, or whatever) that allow to enter their i.e volume, weight during buying. In the end, I would know, that I have 10 bottles with 12 liters in stock, which would be really sufficient. (I do not necessary have to know, if there are 2 bottles with 1.5l, 1 bottle with 2l and 7 bottles with 1l.)

This could also apply to "packs" of goods, where there may sometimes be 10 or 12 or 6 pieces within a pack.

Thanks for reading/considering and kind regards
Dr.A.Colian

**Comment by berrnd** [comment]
> I've done some searching and found topics closely related

True, so I don't really get why you bring up this topic again, talking about the exact same thing every month brings for me only frustration, but that's exactly what I'm used to on this project almost exclusively lately, so let's go on: Just linking one duplicate, guess you've found the other hundreds of threads yourself: #2171

&nbsp;
Just some practical (and here always unpopular) ideas, taken out of reality of using Grocy in a simple way, since life is complex enough:

&nbsp;
> I can buy water in packs, bottles and cans. It can be packed as 10 packs, 4 bottles, 12 cans. A bottle can have 0.75l, 1.5l, 2l, whereas a can can be 0.25l, 0.33l or 0.5l.
> So, especially in comparing prices per liter, it's nice to get the individual amounts in l per packaging unit. Right now, I add all those units manually
- If you want to track each can/bottle individually, while also maintaining the proper relation to "Liter", than a unit like "0.75l Bottle" is required
- Create a proper unit conversion to "Liter", using that as the "Quantity unit for prices", you'll see any price always related to "Liter"
- Buying a Pack of 10 Bottles, don't know what adding another unit "Pack" brings practically - simply track bottles and be surprised that this then fits any Pack size
- Bottle/Can sizes don't change forever and daily, there will be one point in time where all related units will be already present in your Grocy instance (I've created mine years ago and it was probably a 2 minute job, means it was even required that I take my whole annual job vacation time to get that done)

&nbsp;
> I do not necessary have to know, if there are 2 bottles with 1.5l, 1 bottle with 2l and 7 bottles with 1l.

That contradicts itself a little by what you previously pointed out, but that would simply mean:
- Only use a unit "Liter"
- Enter "1.5" on purchase when you purchased one 1.5l bottle or 3 0.5l ones

&nbsp;
Conclusion: All that is already possible, you just need to adjust your personal way of thinking a little bit into how Grocy handles things. Even that I've tried to shape Grocy's stock handling approach pretty open, not forcing a specific workflow, sometimes this is necessary (and just natural for any software on this planet). Since this came up just like thousand times, I've already decided to not change anything about that just like also thousand times - sorry to say that for the 1001st time. If that's the big breaking point for you personally, then I would simply search for another tool. Great that we've talked about that another time in 2023, happy 2024,
